### ISSUE TITLE
perf(segmentation): return an index map instead of 22.7 MB of per-class logits

### ISSUE THREAD

**Issue by msluszniak** [software-mansion/react-native-executorch#1459]
Semantic segmentation returns full-resolution logits for every class. On the 21-class models that is **22.71 MB per inference**, against a 3.24 MB input. The task then reduces those logits away immediately, so nothing downstream ever sees them.

## Evidence that this is the bottleneck

Measured on `deeplab-v3-mobilenet-v3-large`:

- **It barely scales with hardware.** Core ML fp16 runs 41.38 ms on an iPhone SE 3 (A15), 30.59 on an iPhone 16 (A18), 30.08 on an iPhone 17 (A19). That is 1.38x over four generations, where XNNPACK int8 gains 1.92x over the same span. Compute scales, memory bandwidth does not.
- **The compute unit barely matters.** Sweeping all four Core ML compute units moves it 12% end to end (`CPU_AND_NE` 30.64, `ALL` 30.75, `CPU_AND_GPU` 33.77, `CPU_ONLY` 34.42). For comparison, the same sweep on `ssdlite320-mobilenet-v3-large` moves it 75%. If the math were the cost, moving the math would move the number.
- **Both backends pay it equally,** which is why Core ML is only on par with XNNPACK here (30.08 vs 29.39 ms on iPhone 17, the single family in the whole iOS estate where Core ML does not win). Same 520x520 input, same `(1,21,520,520)` output, both fully delegated: Core ML 1 delegate / 0 portable ops, XNNPACK 1 delegate / 1 portable op.

## The logits are already thrown away

`semanticSegmentation.ts` consumes all K channels only to reduce them:

```ts
tOutput
  .copyTo(tReshape)
  .through(toChannelsLast, tChanLast)
  .through(argmax, tMask, -1)          // the 21 channels die here
  .through(applyColormap, tRgba, colormapData);
```

and the public result carries no per-class scores:

```ts
type SemanticSegmentationResult<L> = {
  readonly buffer: ImageBuffer;
  readonly colormap?: ColorMap<L>;
};
```

So moving the argmax into the exported graph costs the caller nothing. It relocates a reduction that already happens, and additionally deletes the `toChannelsLast` transpose over 5.7M floats.

## Proposed change

Export an index map instead of logits: `(1, 520, 520)` int32 rather than `(1, 21, 520, 520)` float32. **22.71 MB -> 1.08 MB, a 21x cut.**

Affected models, all identical in shape:

| model | current output | size |
|---|---|---|
| deeplab-v3-mobilenet-v3-large | (1,21,520,520) f32 | 22.71 MB |
| deeplab-v3-resnet50 | (1,21,520,520) f32 | 22.71 MB |
| deeplab-v3-resnet101 | (1,21,520,520) f32 | 22.71 MB |
| fcn-resnet50 | (1,21,520,520) f32 | 22.71 MB |
| fcn-resnet101 | (1,21,520,520) f32 | 22.71 MB |
| lraspp-mobilenet-v3-large | (1,21,520,520) f32 | 22.71 MB |

`selfie-segmentation` is unaffected: it is single-channel at 256x256 (0.26 MB) and takes the sigmoid path.

## What has to change together

The runtime does not accept this shape today, so the export and the task must land together.

1. `validateSpec` currently pins the contract to `[f32(1,'K','H','W')]` / `[f32('K','H','W')]`. Needs a third variant for an index output.
2. A branch that skips `toChannelsLast` and `argmax` and goes straight to `applyColormap`.
3. The class count has to move into config. `K` is currently read off the output shape, and the `labels.length !== nClasses` guard depends on it.

## Trap to avoid

Do **not** emit the index map as `(1, 1, H, W)` float32. That validates against the existing spec, binds `nClasses = 1`, and silently takes the binary `sigmoid` / `normalize` / `GRAY2RGBA` branch, producing a greyscale ramp of class indices instead of a colour mask. It fails as wrong output, not as an error. An `int32` output fails loudly with `SCHEMA_MISMATCH`, which is the behaviour we want while the two sides are out of step.

## Open question to settle first

`argmax` over the channel dim has to lower to the delegate. If it does not, `lower_full_graph=False` in the segmentation exporters will silently leave it as a portable CPU op over 5.7M floats, which could cost more than the transfer it saves. Cheap to check: probe the exported `.pte` and confirm 1 delegate / 0 portable ops before measuring anything.

A smaller fallback if the contract has to keep logits: emit them as fp16 rather than float32, which halves the output to 11.4 MB for no accuracy the ANE is not already discarding.

## Measurements

iPhone 16 (A18, iOS 26.6.1), release build, RNE benchmark harness, one run per compute unit, pipeline median. Compute unit sweep exported on ExecuTorch 1.4.1 / coremltools 9.0. Device-generation figures are from the v0.10.0 estate runs in #1363.
